### PR TITLE
Helm: Add Clustermesh certificate creation variable

### PR DESCRIPTION
**Pull request by soggiest**
This commit adds the `.Values.clustermesh.apiserver.tls.enableSecrets` variable and new conditional statements to the `tls-provided` Helm templates. The purpose of this is to provide control over secret creation for out-of-band TLS certificate management. The new value defaults to `true` which will maintain the current clustermesh secret creation functionality. If the value is set to `false` Helm will not create TLS secrets for Clustermesh.

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

```release-note
Helm: Add new value `.Values.clustermesh.apiserver.tls.enableSecrets. Setting this value to false will disable the creation of TLS certificate secrets for clustermesh, enabling out-of-band TLS certificate secret management.
```

### CODE REVIEW

File: install/kubernetes/cilium/README.md [hunk 1]
```diff
@@ -227,12 +227,13 @@ contributors across the globe, there is almost always someone available to help.
...
-| clustermesh.apiserver.tls.authMode | string | `"legacy"` | Configure the clustermesh authentication mode. Supported values: - legacy:     All clusters access remote clustermesh instances with the same               username (i.e., remote). The "remote" certificate must be               generated with CN=remote if provided manually. - migration:  Intermediate mode required to upgrade from legacy to cluster               (and vice versa) with no disruption. Specifically, it enables               the creation of the per-cluster usernames, while still using               the common one for authentication. The "remote" certificate must               be generated with CN=remote if provided manually (same as legacy). - cluster:    Each cluster accesses remote etcd instances with a username               depending on the local cluster name (i.e., remote-<cluster-name>).               The "remote" certificate must be generated with CN=remote-<cluster-name>               if provided manually. Cluster mode is meaningful only when the same               CA is shared across all clusters part of the mesh. |
```
Comment (nebril) [inline install/kubernetes/cilium/README.md]: Why are we deleting the docstring here?

**Review by nebril — CHANGES_REQUESTED** [review]
Thanks for the PR! Change looks good overall, I wonder why README.md lost the docstring though.

File: install/kubernetes/cilium/values.yaml [hunk 2]
```diff
@@ -3074,6 +3074,14 @@ clustermesh:
...
       #               if provided manually. Cluster mode is meaningful only when the same
       #               CA is shared across all clusters part of the mesh.
+
+      # -- Allow users to provide their own certificates
+      # Users may need to provide their certificates using 
+      # a mechanism that requires they provide their own secrets.
+      # This setting does not apply to any of the auto-generated 
+      # mechanisms below, it only restricts the creation of secrets
+      # via the `tls-provided` templates.
+      enableSecrets: true
```
Comment (thorn3r) [inline install/kubernetes/cilium/values.yaml]: I think this needs to be moved down below `authMode`. It's cutting out the docstring

**Review by thorn3r — CHANGES_REQUESTED** [review]
left 1 comment, looks good other than that :+1:

File: install/kubernetes/cilium/README.md (hunk 1 above)
Comment (soggiest) [reply install/kubernetes/cilium/README.md]: That was my bad. I've fixed the placement of my change in the values files and updated the docs. This should be resolved now.

File: install/kubernetes/cilium/values.yaml (hunk 2 above)
Comment (soggiest) [reply install/kubernetes/cilium/values.yaml]: Fixed, thanks for pointing that out!

**Review by nebril — APPROVED** [review]
LGTM!

**Review by thorn3r — APPROVED** [review]
excellent :sunglasses:

**Pull request merged**